src/views/Home/Assignment/index.tsx

```diff
@@ -0,0 +1,224 @@
+    contentObjectType: {
+        id: number;
+        title: string;
+    };
+}
+
+interface BulkResponse {
+    assignmentUpdated: number;
+}
+
+interface AssignmentRendererProps extends Assignment {
+    handleClick: (id: number) => void;
+    markAsDonePending: boolean;
+}
+
+const emptyLink = '#';
+const maxItemsPerPage = 5;
+
+function AssignmentRenderer(props: AssignmentRendererProps) {
+    const {
+        id,
+        handleClick: handleClickFromProps,
+        markAsDonePending,
+    } = props;
+    const handleClick = useCallback(() => {
+        handleClickFromProps(id);
+    }, [id, handleClickFromProps]);
+
+    return (
+        <div className={styles.assignmentItem}>
+            <a
+                className={styles.link}
+                href={emptyLink}
+            >
+                {props.createdByDetails.displayName}
+            </a>
+            <span> {_ts('assignment', 'assignedYou')} </span>
+            <a
+                className={styles.link}
+                href={emptyLink}
+            >
+                {props.contentObjectDetails?.title}
+            </a>
+            <span> {_ts('assignment', 'in')} </span>
+            <a
+                href={emptyLink}
+                className={styles.link}
+            >
+                {props.projectDetails?.title}
+            </a>
+            <div className={styles.inline}>
+                <FormattedDate
+                    className={styles.date}
+                    value={props.createdAt}
+                    mode="hh:mm aaa, MMM dd, yyyy"
+                />
+                <Button
+                    transparent
+                    iconName="checkCircle"
+                    className={styles.icon}
+                    onClick={handleClick}
+                    disabled={markAsDonePending}
+                />
+            </div>
+        </div>
+    );
+}
+
+const keySelector = (info: Assignment) => info.id;
+
+
+function Assignments() {
+    const [activePage, setActivePage] = useState<number>(1);
+    const [selectedAssignment, setSelectedAssignment] = useState<number | undefined>();
+    const [
+        pending,
+        assignmentsResponse,
+        ,
+        getAssignments,
+    ] = useRequest<MultiResponse<Assignment>>(
+        {
+            url: 'server://assignments/',
+            method: 'GET',
+            query: {
+                is_done: 3,
+                offset: (activePage - 1) * maxItemsPerPage,
+                limit: maxItemsPerPage,
+            },
+            autoTrigger: true,
+            onFailure: (_, errorBody) =>
+                notifyOnFailure(_ts('assignment', 'assignmentListFetchFailed'))({ error: errorBody }),
+        },
+    );
+
+    const [
+        markAsDonePending,
+        ,
+        ,
+        triggerMarkAsDone,
+    ] = useRequest<MultiResponse<Assignment>>(
+        {
+            url: `server://assignments/${selectedAssignment}/`,
+            method: 'PUT',
+            body: { is_done: true },
```

Comment:
Aren't we going with 1, 2 and 3 here?

Reply:
No. In body is_done has true/false value while in url it takes on integer value of either 1, 2 or 3.